### ISSUE TITLE
Datasets: make less strict

### ISSUE THREAD

**Issue by jepsar** [xdev-software/chartjs-java-model#239]
### Checklist

- [X] I made sure that there are *no existing issues* - [open](https://github.com/xdev-software/chartjs-java-model/issues) or [closed](https://github.com/xdev-software/chartjs-java-model/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have taken the time to fill in all the required details. I understand that the feature request will be dismissed otherwise.
- [X] This issue contains only one feature request/enhancement.

### Description

Currently you can only BarDatasets to BarData. I'm in the process of migrating a project's PrimeFaces 13 data models to XDev data models. There we had a chart with stacked bars and a single line showing the totals. We did it like:
```java
  public void addTotalsLineToBar(final String label) {
    final LineChartDataSet dataSet = new LineChartDataSet();
    final String color = ColorUtils.hslaDark(label);
    dataSet.setData(new ArrayList<>(ChartModels.getTotalsPerLabel(barChart)));
    dataSet.setFill(false);
    dataSet.setLabel(label);
    dataSet.setBackgroundColor(color);
    dataSet.setBorderColor(color);
    dataSet.setTension(0.2);
    barChart.getData().getDataSet().add(0, dataSet);
  }
```
This is with the old PrimeFaces data model which did allow you to add other types of datasets to bar data. Would be great if the XDev model would allow that as well.

**Comment by AB-xdev** [comment]
Thank you for the issue.

We already implemented support for mixed charts with #128 

Here is an example from our tests:
https://github.com/xdev-software/chartjs-java-model/blob/136e904b145a257eda94634291b1d93252234ad5/chartjs-java-model/src/test/java/software/xdev/chartjs/model/BasicChartTest.java#L140-L153
https://github.com/xdev-software/chartjs-java-model/blob/develop/chartjs-java-model/src/test/resources/screenshotReferences/MixedChartBasic.png

Hope this help :)

**Comment by jepsar** [comment]
Awesome! I'll try it like that!
